pypuppetdb/api/__init__.py

```diff
@@ -401,6 +401,10 @@ def nodes(self, unreported=2, with_status=False, **kwargs):
                 else:
                     node['status'] = 'unchanged'
 
+                # Override if available
+                if 'latest_report_status' in node:
+                    node['status'] = node['latest_report_status']
```

Comment:
As per the documentation [1]  ``latest_report_status`` can be one of "failed", "changed" or "unchanged". This code WILL override the noop status.

[1] http://docs.puppetlabs.com/puppetdb/3.2/api/query/v4/nodes.html#query-fields http://docs.puppetlabs.com/puppet/latest/reference/format_report.html#puppettransactionreport